From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gerson Fernando Budke <gerson.budke@ossystems.com.br>
Subject: [PATCH] STM32_ADC_Common: Fix sample dequeuing order

The first sample is zero when using FeedVoltageSampleToChannel. This
fix the correct sample dequeuing order in GetSampleFromChannel. Now
dequeues before the reading to ensure the latest fed sample is used.

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

diff --git a/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs b/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs
--- a/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs
+++ b/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs
@@ @@ private void SwitchToNextChannel()
 
         private uint GetSampleFromChannel(int channelNumber)
         {
-            var sample = sampleProvider[channelNumber].Sample;
             sampleProvider[channelNumber].TryDequeueNewSample();
-            return MillivoltsToSample((double)sample.Value);
+            var sample = sampleProvider[channelNumber].Sample;
+            return MilivoltsToSample((double)sample.Value);
         }
 
         private uint MillivoltsToSample(double sampleInMillivolts)
